src/server/game/Handlers/CharacterHandler.cpp

```diff
@@ -245,7 +245,7 @@ void WorldSession::HandleCharEnum(PreparedQueryResult result)
                 ++num;
             }
         }
-        while (result->NextRow());
+        while (result->NextRow() && num < sWorld->getIntConfig(CONFIG_CHARACTERS_PER_REALM)); // client shows an error if more than 10 characters are listed
```

Comment:
if 10 shows an error, shouldn't this check for 10 (or a const set to 10) instead of a config value ?

Reply:
I did have the same question while typing this. I think both are valid:
CONFIG_CHARACTERS_PER_REALM is enforced to be no more than 10 at config load time.

I'd say it's a game design decision here:
With this right now, if a server owner sets the limit to 5 the player won't see the 6th character either, I think it makes sense to not allow someone to have more character available than defined in any case, but it's really not obvious that the character is just hidden and still exists.
Thus I prefer the constant solution actually. Your call, I can change it and use the constant at config load times as well, 10 is hardcoded there as well.

Reply:
I'd leave the constant 10

Reply:
I would like to have the "10" here and where the config range is enforced to be replaced by a constant defined once and used in both places.

Reply:
Updated now. I wasn't sure where to place the constant though.